### DISCUSSION TITLE
cannot use sqlite with the DebugExecutor

### DISCUSSION THREAD

**Discussion by vito-detullio-finconsgroup** [apache/airflow#21318] [General]
I'm using airflow 1.10.10, and I see in the [DebugExecutor documentation](https://airflow.apache.org/docs/apache-airflow/1.10.10/executor/debug.html):

> Due to its nature the executor can be used with SQLite database.

Unfortunately, writing a simple dag and trying to run produce an error:

```bat
airflow.exceptions.AirflowConfigException: error: cannot use sqlite with the DebugExecutor
```

**Comment by potiuk** [comment]
Don't use Airflow 1.10.10 it's end-of-life as of June 2021. Use airflow 2 instead.

**Reply by vito-detullio-finconsgroup** [reply]
I understand that we are talking about old version. Still: I have ereditated this project, that uses airflow 1.10.10 - and other libraries to to its job.
I tried to upgrade the version of airflow to a newer (1.10.x) one, but I had a lot of troubles with pip. Before I retry to fight the dependencies, can you confirm if I can use the DebugExecutor with sqlite for any 1.10.x version of airflow? or with airflow 2?

**Reply by potiuk** [reply]
No. I can't. 

But using Airflow 1.10 you won't get any support any more anyway - so it's your choice, whether to stay alone with your problem or migrate to version that even if it maybe has problems, gets maintenance, security fixes and generally welcoming and helpful community.

BTW. Yes. `pip` introduced breaking changes and as you can see on the installation page of our 1.10.15 (bridge release to migrate to Airlfow 2) you need to use old version of PIP to install earlier versions of Airflow

https://airflow.apache.org/docs/apache-airflow/1.10.15/installation.html 

> On November 2020, new version of PIP (20.3) has been released with a new, 2020 resolver. This resolver does not yet work with Apache Airflow and might leads to errors in installation - depends on your choice of extras. In order to install Airflow you need to either downgrade pip to version 20.2.4 pip upgrade --pip==20.2.4 or, in case you use Pip 20.3, you need to add option --use-deprecated legacy-resolver to your pip install command.

**Reply by vito-detullio-finconsgroup** [reply]
FWIW, according to [the airflow documentation](https://airflow.apache.org/docs/apache-airflow/stable/installation/supported-versions.html) even the airflow 2 version should be in "limited support" since Dec 2021...

**Reply by potiuk** [reply]
> FWIW, according to [the airflow documentation](https://airflow.apache.org/docs/apache-airflow/stable/installation/supported-versions.html) even the airflow 2 version should be in "limited support" since Dec 2021...

The documentation is "per-version" and this part is already out-dated (it was not updated in the 2.2.3 documentation when we released it). It is alreaedy corrected in the upcoming 2.2.4 (in ~ 2 weeks) https://github.com/apache/airflow/blob/v2-2-test/docs/apache-airflow/installation/supported-versions.rst 

And since this (like any "live"project) keeps on changing, the authoritative source for that is only here (in our GitHub Repo): https://github.com/apache/airflow/blob/main/README.md#version-life-cycle
